test/module/shared_model/builders/protobuf/transport_builder_test.cpp

```diff
@@ -475,77 +450,68 @@ TEST_F(TransportBuilderTest, BlockVariantWithInvalidBlock) {
-                               validation::FieldValidator,
-                               validation::CommandValidatorVisitor<
-                                   validation::FieldValidator>>,
-                           validation::BatchOrderValidator>>()
-          .build(tr));
+  auto val = framework::expected::val(TransactionSequenceBuilder().build(tr));
   ASSERT_TRUE(val);
-  val | [](auto &seq) { ASSERT_EQ(boost::size(seq.value.transactions()), 0); };
+  val | [](auto &seq) { EXPECT_EQ(boost::size(seq.value.transactions()), 0); };
 }
 
+struct getProtocolTx {
+  iroha::protocol::Transaction operator()(
+      const std::shared_ptr<interface::Transaction> tx) const {
+    return std::static_pointer_cast<proto::Transaction>(tx)->getTransport();
+  }
+};
+
 /**
  * @given sequence of transaction with a right order
  * @when TransportBuilder tries to build TransactionSequence object
  * @then  built object contains TransactionSequence shared model object
  */
 TEST_F(TransportBuilderTest, TransactionSequenceCorrect) {
-  std::vector<proto::Transaction> transactions;
-  auto batch1 = getValidBatch(0, 10);
-  auto batch2 = getValidBatch(20, 5);
-  auto batch3 = getValidBatch(30, 5);
+  std::vector<std::shared_ptr<interface::Transaction>> transactions;
+  auto now = iroha::time::now();
+  auto batch1 = framework::batch::createUnsignedBatchTransactions(
+      interface::types::BatchType::ATOMIC, 10, now);
+  auto batch2 = framework::batch::createUnsignedBatchTransactions(
+      interface::types::BatchType::ATOMIC, 5, now + 1);
+  auto batch3 = framework::batch::createUnsignedBatchTransactions(
+      interface::types::BatchType::ATOMIC, 5, now + 2);
   std::move(
       std::begin(batch1), std::end(batch1), std::back_inserter(transactions));
   std::move(
       std::begin(batch2), std::end(batch2), std::back_inserter(transactions));
-  transactions.push_back(createTransaction());
-  transactions.push_back(createTransaction());
-  transactions.push_back(createTransaction());
+  transactions.emplace_back(clone(createTransaction()));
+  transactions.emplace_back(clone(createTransaction()));
+  transactions.emplace_back(clone(createTransaction()));
   std::move(
       std::begin(batch3), std::end(batch3), std::back_inserter(transactions));
-  transactions.push_back(createTransaction());
+  transactions.emplace_back(clone(createTransaction()));
+
+  auto protocol_transactions =
+      transactions | boost::adaptors::transformed(getProtocolTx());
   auto val = framework::expected::val(
-      TransactionSequenceBuilder().build(transactions));
+      TransactionSequenceBuilder().build(protocol_transactions));
   ASSERT_TRUE(val);
   val | [](auto &seq) { ASSERT_EQ(boost::size(seq.value.transactions()), 24); };
 }
-/**
- * @given batch of transaction with transaction in the middle
- * @when TransportBuilder tries to build TransactionSequence object
- * @then  built an error
- */
-TEST_F(TransportBuilderTest, TransactionInteraptedBatch) {
```

Comment:
Drop a couple of words in reply to that comment why the test is not needed anymore.

Reply:
This case is handled by fabric method in create transaction sequence. It automatically sort transactions into batches